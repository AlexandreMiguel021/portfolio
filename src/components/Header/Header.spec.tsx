import { render, fireEvent } from '@testing-library/react'
import { Header } from '.'
import '@testing-library/jest-dom'

describe('Header', () => {
	it('should render all urls correctly', () => {
		const { getByText } = render(<Header />)
		expect(getByText(/projetos/i).closest('a')).toHaveAttribute(
			'href',
			'/#projects'
		)
		expect(getByText(/habilidades/i).closest('a')).toHaveAttribute(
			'href',
			'/#skills'
		)
		expect(getByText(/sobre mim/i).closest('a')).toHaveAttribute(
			'href',
			'/#about'
		)
		expect(getByText(/contato/i).closest('a')).toHaveAttribute(
			'href',
			'/#contact'
		)
	})

	it('should render a Header correctly', () => {
		const { container } = render(<Header />)
		expect(container).toMatchSnapshot()
	})

	it('should open and close a mobile menu if the button icon is clicked', () => {
		const { getByTestId, container } = render(<Header />)
		fireEvent.click(getByTestId('menu'))
		expect(container.querySelector('nav')).toHaveClass('active')
		fireEvent.click(getByTestId('menu'))
		expect(container.querySelector('nav')).toHaveClass('menu')
	})
})
