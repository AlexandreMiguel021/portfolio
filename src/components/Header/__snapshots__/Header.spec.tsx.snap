// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render a Header correctly 1`] = `
<div>
  <header
    class="header"
    id="header"
  >
    <div
      class="wrapper"
    >
      <a
        class="logo"
        data-aos="flip-up"
        href="/"
      >
        <span>
          Alexandre
        </span>
      </a>
      <nav
        class="menu"
      >
        <ul
          class="items"
        >
          <li
            class="item"
          >
            <a
              class="link"
              href="/#skills"
            >
              Habilidades
            </a>
          </li>
          <li
            class="item"
          >
            <a
              class="link"
              href="/#about"
            >
              Sobre mim
            </a>
          </li>
          <li
            class="item"
          >
            <a
              class="link"
              href="/#projects"
            >
              Projetos
            </a>
          </li>
          <li
            class="item"
          >
            <a
              class="link"
              href="/#contact"
            >
              Contato
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="menuIcon"
        data-testid="menu"
      >
        <svg
          aria-label="Abrir"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.5 12a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </header>
  <div
    class="spacer"
  />
</div>
`;
